v3.5.0 OpenSubdivConfig.cmake usage

I am building a simple example to learn about the cmake config file shipped with v3.5.0

I can't seem to get it to include the include path

Error output
```
[ 50%] Building CXX object CMakeFiles/check_linkage.dir/check_linkage.cpp.o
/data2/nyue/projects/OpenSubdiv/OpenSubdivBfr/OpenSubdivBfr_git/check_linkage.cpp:1:10: fatal error: opensubdiv/far/topologyRefiner.h: No such file or directory
 #include <opensubdiv/far/topologyRefiner.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
CMakeFiles/check_linkage.dir/build.make:75: recipe for target 'CMakeFiles/check_linkage.dir/check_linkage.cpp.o' failed
make[2]: *** [CMakeFiles/check_linkage.dir/check_linkage.cpp.o] Error 1
CMakeFiles/Makefile2:82: recipe for target 'CMakeFiles/check_linkage.dir/all' failed
make[1]: *** [CMakeFiles/check_linkage.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2
```

CMakeLists.txt
```
cmake_minimum_required(VERSION 3.12)

project(bfr)

find_package(OpenSubdiv CONFIG REQUIRED)

add_executable(check_linkage
  check_linkage.cpp
  )

target_link_libraries(check_linkage
  OpenSubdiv::osdCPU_static
  OpenSubdiv::osdGPU_static
  )
```

Source file
```
#include <opensubdiv/far/topologyRefiner.h>
#include <opensubdiv/bfr/refinerSurfaceFactory.h>
#include <opensubdiv/bfr/surface.h>

#include <vector>
#include <string>
#include <cstring>
#include <cstdio>

int main()
{
	return 0;
}
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -846,6 +846,7 @@ endmacro()
 #-------------------------------------------------------------------------------
 # Build targets
 
+include(GNUInstallDirs)
 
 # if you want to build examples against installed OpenSubdiv header files,
 # use OPENSUBDIV_INCLUDE_DIR.
@@ -887,34 +888,31 @@ if (NOT NO_DOC)
     add_subdirectory(documentation)
 endif()
 
-if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.2)
-    #
-    # CMake Config.
-    #
-    include(GNUInstallDirs)
-    include(CMakePackageConfigHelpers)
+#
+# CMake Config.
+#
+include(CMakePackageConfigHelpers)
 
-    set(OPENSUBDIV_CONFIG_PATH "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+set(OPENSUBDIV_CONFIG_PATH "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
 
-    configure_package_config_file(
-        opensubdiv-config.cmake.in
-        ${CMAKE_CURRENT_BINARY_DIR}/OpenSubdivConfig.cmake
-        INSTALL_DESTINATION ${OPENSUBDIV_CONFIG_PATH}
-    )
-    write_basic_package_version_file(
-        ${CMAKE_CURRENT_BINARY_DIR}/OpenSubdivConfigVersion.cmake
-        VERSION ${OSD_SONAME}
-        COMPATIBILITY SameMajorVersion
-    )
+configure_package_config_file(
+    opensubdiv-config.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/OpenSubdivConfig.cmake
+    INSTALL_DESTINATION ${OPENSUBDIV_CONFIG_PATH}
+)
+write_basic_package_version_file(
+    ${CMAKE_CURRENT_BINARY_DIR}/OpenSubdivConfigVersion.cmake
+    VERSION ${OSD_SONAME}
+    COMPATIBILITY SameMajorVersion
+)
 
-    install(EXPORT opensubdiv-targets
-        NAMESPACE OpenSubdiv::
-        FILE OpenSubdivTargets.cmake
-        DESTINATION ${OPENSUBDIV_CONFIG_PATH})
+install(EXPORT opensubdiv-targets
+    NAMESPACE OpenSubdiv::
+    FILE OpenSubdivTargets.cmake
+    DESTINATION ${OPENSUBDIV_CONFIG_PATH})
 
-    install(FILES
-        ${CMAKE_CURRENT_BINARY_DIR}/OpenSubdivConfig.cmake
-        ${CMAKE_CURRENT_BINARY_DIR}/OpenSubdivConfigVersion.cmake
-        DESTINATION ${OPENSUBDIV_CONFIG_PATH}
-    )
-endif()
+install(FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/OpenSubdivConfig.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/OpenSubdivConfigVersion.cmake
+    DESTINATION ${OPENSUBDIV_CONFIG_PATH}
+)

--- a/opensubdiv/CMakeLists.txt
+++ b/opensubdiv/CMakeLists.txt
@@ -141,6 +141,11 @@ if (NOT NO_LIB)
             FOLDER "opensubdiv"
     )
 
+    target_include_directories(osd_static_cpu
+        INTERFACE
+            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+    )
+
     target_link_libraries(osd_static_cpu
         ${PLATFORM_CPU_LIBRARIES}
     )
@@ -162,6 +167,11 @@ if (NOT NO_LIB)
                 OUTPUT_NAME osdGPU
                 EXPORT_NAME osdGPU_static
                 CLEAN_DIRECT_OUTPUT 1)
+
+        target_include_directories(osd_static_gpu
+            INTERFACE
+                $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+        )
 
         target_link_libraries(osd_static_gpu
             ${PLATFORM_CPU_LIBRARIES} ${PLATFORM_GPU_LIBRARIES}
@@ -204,6 +214,11 @@ if (NOT NO_LIB)
                 )
         endif()
 
+        target_include_directories(osd_dynamic_cpu
+            INTERFACE
+                $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+        )
+
         target_link_libraries(osd_dynamic_cpu
             ${PLATFORM_CPU_LIBRARIES}
         )
@@ -236,6 +251,11 @@ if (NOT NO_LIB)
                         CLEAN_DIRECT_OUTPUT 1
                     )
             endif()
+
+            target_include_directories(osd_dynamic_gpu
+                INTERFACE
+                    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+            )
 
             target_link_libraries(osd_dynamic_gpu
                 osd_dynamic_cpu

